src/core/error_handling.ts

```diff
@@ -127,6 +127,42 @@ API.fatal_error = function (e: any) {
   throw e;
 };
 
+class FatalPyodideError extends Error {}
+Object.defineProperty(FatalPyodideError.prototype, "name", {
+  value: FatalPyodideError.name,
+});
+
+let stderr_chars: number[] = [];
+API.capture_stderr = function () {
+  stderr_chars = [];
+  const FS = Module.FS;
+  FS.createDevice("/dev", "capture_stderr", null, (e: number) =>
+    stderr_chars.push(e)
+  );
+  FS.closeStream(2);
+  FS.open("/dev/capture_stderr", 1);
+};
+
+API.restore_stderr = function () {
+  const FS = Module.FS;
+  FS.closeStream(2);
+  FS.unlink("/dev/capture_stderr");
+  FS.open("/dev/stderr", 1);
```

Comment:
I guess [`1` is `O_WRONLY`](https://github.com/emscripten-core/emscripten/blob/e05e72d9c49fe15578e73934ce525a894d1b712a/system/lib/libc/musl/include/fcntl.h#L49)? Could you add a comment on that or it will be hard to understand in a first glance.